File: Dr.NutrizioNino.Models/Dto/RecipeDetailDto.cs
namespace Dr.NutrizioNino.Models.Dto;

public record RecipeDetailNutrientDto(
    Guid NutrientId,
    string Name,
    int PositionOrder,
    Guid UnitOfMeasureId,
    decimal Quantity
);

public record RecipeDetailIngredientDto(
    Guid FoodId,
    string FoodName,
    decimal QuantityGrams
);

public record RecipeDetailDto(
    Guid Id,
    string Name,
    decimal WeightGrams,
    IList<RecipeDetailIngredientDto> Ingredients,
    IList<RecipeDetailNutrientDto> Nutrients
);

public record RescaleRecipeRequest(decimal WeightGrams, bool Recalculate = false);

File: src/Dr.NutrizioNino.WebVue/src/Interfaces/recipes/RecipeDetailDto.ts
export interface RecipeDetailNutrientDto {
  nutrientId: string
  name: string
  positionOrder: number
  unitOfMeasureId: string
  quantity: number
}

export interface RecipeDetailIngredientDto {
  foodId: string
  foodName: string
  quantityGrams: number
}

export interface RecipeDetailDto {
  id: string
  name: string
  weightGrams: number
  ingredients: RecipeDetailIngredientDto[]
  nutrients: RecipeDetailNutrientDto[]
}
